ui/pages/manage_connected_accounts.tsx

```diff
@@ -0,0 +1,280 @@
+    };
+  }
+
+  return Object.values(byProvider);
+};
+
+const ManageConnectedAccounts: NextPage = () => {
+  const [flow, setFlow] = useState<SettingsFlow>();
+
+  const router = useRouter();
+  const toastNotify = useToastNotification();
+  const { return_to: returnTo, flow: flowId } = router.query;
+
+  const userName = getLoggedInName(flow);
+  const providers = useMemo(() => buildOidcProviderStates(flow), [flow]);
+
+  useEffect(() => {
+    if (!router.isReady || providers.length === 0) return;
+
+    const pendingProvider = window.sessionStorage.getItem(
+      "pending_provider_link",
+    );
+    if (!pendingProvider) return;
+
+    const isLinked = providers.some(
+      (provider) =>
+        provider.id === pendingProvider && provider.name === "unlink",
+    );
+
+    if (isLinked) {
+      toastNotify.success(
+        `Your ${pendingProvider} account is now connected and can be used to sign in.`,
+        undefined,
+        "Account connected successfully",
+      );
+    }
+
+    window.sessionStorage.removeItem("pending_provider_link");
+  }, [router.isReady, providers]);
+
+  useEffect(() => {
+    if (!router.isReady || flow) {
+      return;
+    }
+
+    if (flowId) {
+      kratos
+        .getSettingsFlow({ id: String(flowId) })
+        .then((res) => setFlow(res.data))
+        .catch(handleFlowError("settings", setFlow));
+      return;
+    }
+
+    kratos
+      .createBrowserSettingsFlow({
+        returnTo: returnTo ? String(returnTo) : undefined,
+      })
+      .then(({ data }) => {
+        if (flowId !== data.id) {
+          window.history.replaceState(
+            null,
+            "",
+            `./manage_connected_accounts?flow=${data.id}`,
+          );
+          router.query.flow = data.id;
+        }
+        setFlow(data);
+      })
+      .catch(async (err: AxiosError<string>) => {
+        if (err.response?.data.trim() === "Failed to create settings flow") {
+          window.location.href = `./login?return_to=${window.location.pathname}`;
+          return;
+        }
+
+        return Promise.reject(err);
+      });
+  }, [flowId, router, router.isReady, returnTo, flow]);
+
+  const handleSubmit = useCallback(
+    (values: UpdateSettingsFlowBody) => {
+      const methodValues = values as UpdateSettingsFlowWithOidcMethod;
+      return kratos
+        .updateSettingsFlow({
+          flow: String(flow?.id),
+          updateSettingsFlowBody: {
+            method: "oidc",
+            ...(methodValues.link ? { link: methodValues.link } : {}),
+            ...(methodValues.unlink ? { unlink: methodValues.unlink } : {}),
+          },
+        })
+        .then((result) => {
+          const data = result.data as SettingsFlowWithRedirect;
+          if (data.redirect_to) {
+            const providerId = methodValues.link ?? methodValues.unlink ?? "";
+            if (providerId) {
+              sessionStorage.setItem("pending_provider_link", providerId);
```

Comment:
question: how does this work? Here we use `sessionStorage`, but in the rest of the code we use `window.sessionStorage`. When I try to check my firefox storage I cannot this cookie being set (if you use `window.sessionStorage.setItem` then it will show up). I would assume that react or nextjs handles this.

Reply:
Linking to the other discussion, maybe we should rely on the return paramter instead of using session storage? See also https://github.com/canonical/identity-platform-login-ui/pull/731#discussion_r2318191785